src/fabric/src/fabric2_fdb.erl

```diff
@@ -346,12 +350,68 @@ delete(#{} = Db) ->
     } = ensure_current(Db),
 
     DbKey = erlfdb_tuple:pack({?ALL_DBS, DbName}, LayerPrefix),
-    erlfdb:clear(Tx, DbKey),
-    erlfdb:clear_range_startswith(Tx, DbPrefix),
+    DoRecovery = fabric2_util:do_recovery(),
+    case DoRecovery of
+        true ->
+            Timestamp = list_to_binary(fabric2_util:iso8601_timestamp()),
+            DeletedDbKey = erlfdb_tuple:pack({?DELETED_DBS, DbName, Timestamp},
+                LayerPrefix),
+            erlfdb:set(Tx, DeletedDbKey, DbPrefix),
```

Comment:
It just occurred to me, we need to read the `DeletedDbKey` and make sure that it comes back `not_found` or else we end up losing references to deleted databases if users create/delete databases too quickly.

Reply:
thanks @davisp for bringing up this corner case. I added some lines to make check. Also I made change on API naming. Could you please review again?